backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/DockerCassandraRule.java

```diff
@@ -48,8 +48,8 @@ public Statement apply(Statement base, Description description) {
 
     public void start() {
         DockerCassandraSingleton.singleton.start();
-        DockerCassandraSingleton.incrementTestsPlayed();
         if (allowRestart) {
+            DockerCassandraSingleton.incrementTestsPlayed();
```

Comment:
That's wrong: we always want to count the number of tests since last restart

Reply:
Even for the classes we don't want a restart with in the first place?

Reply:
Can't we have a method to just "increment?". Then we can restart only when restart are allowed.